lib/nerves_hub_cli/user.ex

```diff
@@ -27,14 +34,13 @@ defmodule NervesHubCLI.User do
     end
   end
 
-  @spec auth(String.t()) :: {:error, atom()} | {:ok, %{cert: binary(), key: binary()}}
+  @spec auth(String.t()) :: {:error, atom()} | {:ok, auth_map}
   def auth(password) do
     with {:ok, encrypted} <- File.read(user_data_path(@key)),
          {:ok, pem_key} <- Crypto.decrypt(encrypted, password),
-         {:ok, pem_cert} <- File.read(user_data_path(@cert)) do
-      key = Certificate.pem_to_der(pem_key)
-      cert = Certificate.pem_to_der(pem_cert)
-
+         {:ECPrivateKey, _, _, _, _} = key <- PrivateKey.from_pem(pem_key),
```

Comment:
Ugh - I hate these pattern matches here. Voltone is fixing his API to return `{:ok, rc}` tuples, but it's not out yet. I'm personally fine with the bang versions, since 1. these conversions shouldn't fail if decryption doesn't fail and 2. the error tuple from `X509.PrivateKey.from_pem` is not much more illuminating than "fail".